pkg/signatures/factory.go

```diff
@@ -113,6 +119,11 @@ func createVerifiersFromIntegration(integration *storage.SignatureIntegration) [
 // It will retry on transient errors and return the fetched signatures.
 func FetchImageSignaturesWithRetries(ctx context.Context, fetcher SignatureFetcher, image *storage.Image,
 	fullImageName string, registry registryTypes.Registry) ([]*storage.Signature, error) {
+	// Short-circuit if signature fetching is disabled.
+	if env.DisableSignatureFetching.BooleanSetting() {
```

Comment:
Perhaps by moving it to `enricher_impl` we could've avoid some extra-work in central

Reply:
Yeah. However, we have to ensure that local scanning is also respecting the environment variable, hence adding it only here means we won't have to think about all the use-cases where signature fetching is used, so I'd prefer it here tbh.